src/bp/core/services/auth/auth-service.ts

```diff
@@ -56,23 +55,53 @@ export default class AuthService {
       throw new PasswordExpiredError()
     }
 
-    return user.id
+    return user.email
+  }
+
+  async createBasicUser(user: Partial<BasicAuthUser>): Promise<CreatedUser> {
+    const newUser = {
+      ...user
+    } as BasicAuthUser
+
+    const createdUser = await this.workspace.createUser(newUser)
+    return {
+      password: user.password ? user.password : await this.resetPassword(user.email!),
+      user: createdUser
+    }
+  }
+
+  async createExternalUser(user: Partial<ExternalAuthUser>, provider: string): Promise<CreatedUser> {
```

Comment:
what is provider ? does it give more information then the used auth strategy ? or is it the same ? if yes then I think it should be typed

Reply:
provider is essentially the strategy name

Reply:
then it should be typed

Reply:
you're right, it's fixed